backend/prompt_studio/prompt_studio_core_v2/models.py

```diff
@@ -0,0 +1,158 @@
+import logging
+import shutil
+import uuid
+from typing import Any
+
+from account_v2.models import User
+from adapter_processor_v2.models import AdapterInstance
+from django.db import models
+from django.db.models import QuerySet
+from file_management.file_management_helper import FileManagerHelper
+from prompt_studio.prompt_studio_core_v2.constants import DefaultPrompts
+from utils.models.base_model import BaseModel
+from utils.models.organization_mixin import (
+    DefaultOrganizationManagerMixin,
+    DefaultOrganizationMixin,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class CustomToolModelManager(DefaultOrganizationManagerMixin, models.Manager):
+
+    def for_user(self, user: User) -> QuerySet[Any]:
+        return (
+            self.get_queryset()
+            .filter(models.Q(created_by=user) | models.Q(shared_users=user))
+            .distinct("tool_id")
+        )
+
+
+class CustomTool(DefaultOrganizationMixin, BaseModel):
```

Comment:
@muhammad-ali-e I notice that we set the org_id by referring the `StateStore`. Where / how will this be set in case of deployed pipeline runs?

Reply:
@muhammad-ali-e on pipeline run